streams/src/main/java/org/apache/kafka/streams/processor/assignment/TaskAssignmentUtils.java

```diff
@@ -72,6 +80,27 @@ public static Map<ProcessId, KafkaStreamsAssignment> identityAssignment(final Ap
         return assignments;
     }
 
+    /**
+     * Assign standby tasks to KafkaStreams clients according to the default logic.
+     * <p>
+     * If rack-aware client tags are configured, the rack-aware standby task assignor will be used
+     *
+     * @param applicationState        the metadata and other info describing the current application state
+     * @param kafkaStreamsAssignments the current assignment of tasks to KafkaStreams clients
+     *
+     * @return a new map containing the mappings from KafkaStreamsAssignments updated with the default standby assignment
+     */
+    public static Map<ProcessId, KafkaStreamsAssignment> defaultStandbyTaskAssignment(final ApplicationState applicationState,
+                                                                                      final Map<ProcessId, KafkaStreamsAssignment> kafkaStreamsAssignments) {
+        if (!applicationState.assignmentConfigs().rackAwareAssignmentTags().isEmpty()) {
+            return tagBasedStandbyTaskAssignment(applicationState, kafkaStreamsAssignments);
+        } else if (canPerformRackAwareOptimization(applicationState, AssignedTask.Type.STANDBY)) {
+            return tagBasedStandbyTaskAssignment(applicationState, kafkaStreamsAssignments);
```

Comment:
Address in a followup:

We should just remove this case entirely right? basically it's "if hasRackAwareTags then do tag-based standby task assignment, if doesNotHaveTags then do default standby task assignment"

Note that the tag-based rack aware assignment has nothing to do with the rack ids. So `canPerformRackAwareOptimization` is kind of irrelevant to the question here